process/accountProcessor.go

```diff
@@ -452,3 +463,11 @@ func (ap *AccountProcessor) IsDataTrieMigrated(address string, options common.Ac
 
 	return nil, ErrSendingRequest
 }
+
+func getAvailabilityBasedOnAccountQueryOptions(options common.AccountQueryOptions) data.ObserverDataAvailabilityType {
```

Comment:
I would create a stateless component for these If's because the ifs are higher level decision-making code
The component could also have functions that will be used in nodesHolder component L58
L157
L182

Reply:
done